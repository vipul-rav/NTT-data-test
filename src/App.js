import React, { memo } from 'react';
import { AppRouter } from './routes';
import { ErrorComponent } from './components/Error';

const AppComponent = memo(function App() {
  return (
    <>
      <div>NTT Data Test</div>
      <ErrorComponent>{AppRouter}</ErrorComponent>
    </>
  );
});

const App = AppComponent;

export { App };
