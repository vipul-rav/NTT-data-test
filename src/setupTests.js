import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

const AllTheProviders = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
