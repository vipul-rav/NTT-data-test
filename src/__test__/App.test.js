import React from 'react';
import { cleanup } from '@testing-library/react';
import { render } from '../setupTests';
import { App } from '../App';

const setup = (overridesProps) => {
  const props = {
    ...overridesProps
  };

  const { container, getByText } = render(<App {...props} />);

  return {
    props,
    container,
    getByText
  };
};

describe('App', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
