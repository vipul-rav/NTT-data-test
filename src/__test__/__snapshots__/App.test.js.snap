// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div>
  <div>
    NTT Data Test
  </div>
  <div>
    Loading...
  </div>
</div>
`;
